fuzzers/fuzzbench/src/lib.rs

```diff
@@ -364,7 +364,7 @@ fn fuzz(
     // Autodict Tokens
     #[cfg(target_os = "linux")]
     {
-        let autodict_tokens = unsafe { Tokens::from_autodict(TOKEN_START, TOKEN_STOP)? };
+        let autodict_tokens = unsafe { Tokens::from_autotokens(TOKEN_START, TOKEN_STOP)? };
```

Comment:
Still don't like the API :)
Either have a `autotokens()` fn targets that returns them or have this fn without parameters if that's somehow possible

Reply:
The real problem is that https://github.com/AFLplusplus/LibAFL/blob/1eb571b2aad64fe2e803f63d09c14775660fbdf9/fuzzers/fuzzbench/src/lib.rs#L360 are removed when this second instance is added

Reply:
so we need something like

```rust
state.add_metadata(
  Tokens::builder()
    .parse_tokens_file(tokenfile)?
    .from_autotokens(TOKEN_START, TOKEN_STOP)?
    .build()
); 
```

Reply:
> Still don't like the API :) Either have a `autotokens()` fn targets that returns them or have this fn without parameters if that's somehow possible

No it makes sense to me to have pointer args because you can have another source of autotokens, like a shmem for the forkserver executor

Reply:
I'm fine with an additional raw API but it shouldn't be the one users commonly use imho. Even if we have support for forkserver tokens at some point they should get a nice api

Reply:
at least I can
```
+#[cfg(target_os = "linux")]
+pub unsafe fn token_section() -> (*const u8, *const u8) {
+    (__token_start, __token_stop)
+}
```
this to make it less ugly

Reply:
High-level I'd like something like
`map.entry::<Tokens>().or_insert(default).append(libafl_target::autotokens())`

Reply:
Now it is wrapped in a struct
for the most frequently used pointer from elf section, we can use
```
            let token_section = unsafe { TokenSection::new(token_section()) };
```
or it can be constructed from any const u8 pointer like
```
 let x = unsafe {TokenSection::new(ptr1, ptr2)}
```

Reply:
A lot better like this, thx. 
Now I think we should get rid of unsafe: 
- only Tokens created from a raw pointer need to have an unsafe constructor
- the autotokens should either contain tokens, or not (or even return` None`?), but always be safe, as long as they use our compiler passes
- likewise, tokens read from a file are safe

Reply:
actually `libafl` code (to be more specific struct `Tokens` and `TokenSection` ) does not have any unsafe code.
only token_mutations() in libafl_targets needs it (cuz it returns raw ptrs)

the above code can also be written as
 `            let token_section = TokenSection::new(unsafe { token_section() }) ;`
to make it clear.

Reply:
wait, no

Reply:
Why would tokens_section need to be unsafe?
It's either set or or isn't, the user shouldn't care

Reply:
Finally got it. I was confused with Rust's unsafe semantics😅